### DISCUSSION TITLE
Migrate a traditional plugin to block-based

### DISCUSSION THREAD

**Discussion by mburridge** [WordPress/developer-blog-content#42] [Topic Ideas]
A post detailing some strategies to migrate a traditional PHP-based plugin that renders content (e.g. shortcode, widget) to be block-based instead, but without losing or replacing the traditional functionality.

This post should outline the reasons and benefits of doing so, as well as some technical considerations that should be taken into account during the process of adding block functionality to a traditional plugin, with some code samples.

**Comment by bph** [comment]
Isn't there a way to cover the topic [via the Case Study? ](https://github.com/WordPress/developer-blog-content/discussions/44) 

If the topic warrants more than one blog post, and I am sure it does, the title for each of the two can be more specific and the description be more specific on the covered areas.

**Comment by justintadlock** [comment]
I was thinking of making a separate topic based on [this forum post](https://wordpress.org/support/topic/create-gutenberg-widget/#post-16565049). But, I wanted to jump back to this discussion first.

The developer in the linked thread has created PHP widgets in the past but is struggling to turn them into blocks. 

We have a tutorial on [converting shortcodes](https://developer.wordpress.org/news/2023/03/converting-your-shortcodes-to-blocks/), which I think is a decent foundation. However, I think we need some beefier examples of utilizing custom InspectorControls (converting widget/shortcode options to controls in the block inspector).

I should have dozens of old widgets lying around that I could build out something for a step-by-step walkthrough of doing this.
